### ISSUE TITLE
DateRangePicker is difficult to type in

### ISSUE THREAD

**Issue by acgrant** [mui/mui-x#5351]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Current behavior 😯

Hello, I have the MUI Pro Plan for my company. Typing into the text field for the MUI DateRangePicker is not as smooth as I would hope. I've noticed this is my view and even in the demo on the MUI site. For example, typing "02/22/2022" results in "22/22/022." It appears that the first zero is disregarded. One can get around that; however, it starts to degrade even further when the year is typed, almost rendering that section unusable manually.

<img width="369" alt="Screen Shot 2022-06-29 at 16 22 40" src="https://user-images.githubusercontent.com/102261834/176561955-38580c79-05bc-4f44-bc4b-e37ab4691f6a.png">

### Expected behavior 🤔

I want to be able to type into the MUI DateRangePicker without an issue.

### Steps to reproduce 🕹

Steps:

1. Type (don't paste) "02/22/2022" one key at a time into the text field of the DateRangePicker.

### Context 🔦

I want to type into the TextField for the DateRangePicker and have it register a date. I'm on Chrome, using and Mac with an M1 chip. I've experienced this on my site and even in the demo https://mui.com/x/react-date-pickers/date-range-picker/#basic-usage.

### Your environment 🌎

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
  Don't forget to mention which browser you used.
  Output from `npx @mui/envinfo` goes here.
```
</details>

**Comment by alexfauquette** [comment]
This looks like a duplicate of #5284 which get fixed by #5265

Don't hesitate to reopen the issue if it still appears after the next release

**Comment by acgrant** [comment]
@alexfauquette I'm not sure how to reopen the issue on here if it was closed by a collaborator of the repo; however, I'm still experiencing the issue. The update did improve things; however, it's still not registering the first digit that's entered and it starts really breaking down when the year is typed. Here is what it looks like when I type "12252022"
![Screen Shot 2022-07-28 at 10 19 49](https://user-images.githubusercontent.com/102261834/181598835-b4896146-bf38-4d26-b19b-a72965ee4f7c.png)

**Comment by alexfauquette** [comment]
With the last version of the pickers?

I tried to do some edits using the documentation and it seems to work fine

[Screencast from 29. 07. 22 09:25:28.webm](https://user-images.githubusercontent.com/45398769/181707134-7b13a555-2b70-4212-bd8f-c4fdb6e75168.webm)

**Comment by acgrant** [comment]
@alexfauquette Thank you for your response and the video. I'll look into how I can reproduce this for you, promptly.

**Comment by acgrant** [comment]
@alexfauquette I have 5.0.0-alpha.7 currently for the Date Range Picker.


https://user-images.githubusercontent.com/102261834/182479171-1f38e688-6f72-4226-a58b-8bd1319610b2.mp4


Here's the code for the section (DateRangeTextFields is just a component composed of two TextFields separated by a Typography element), let me know what else could be affecting this behavior if it's not reproducible on your side or if there's any part of my side you'd like me to elaborate on: 
```js
<LocalizationProvider dateAdapter={AdapterDateFns} localeText={{ start: 'Start date', end: 'End date' }}>
  <DateRangePicker
    components={{ OpenPickerIcon: DateRangeIcon }}
    onChange={(dateRange): void => {
      grid.setSelectedDateRange(dateRange);
      setFilterTable((prevState) => !prevState);
    }}
    onClose={(): void => setDateRangePickerOpen(false)}
    onOpen={(): void => setDateRangePickerOpen(true)}
    open={dateRangePickerOpen}
    renderInput={(startProps, endProps): JSX.Element => (
      <DateRangeTextFields startProps={startProps} endProps={endProps} toggler={dateRangePickerOpenHandler} />
    )}
    value={grid.selectedDateRange}
  />
</LocalizationProvider>
```

**Comment by alexfauquette** [comment]
#5265 fixed range picker bugs about the keyboard input. The fix has been released in `v5.00-beta.0`

**Comment by acgrant** [comment]
Got you, that was a big help, thank you @alexfauquette
